internal/controllers/machinedeployment/machinedeployment_controller.go

```diff
@@ -83,10 +82,6 @@ func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, opt
 	err = ctrl.NewControllerManagedBy(mgr).
 		For(&clusterv1.MachineDeployment{}).
 		Owns(&clusterv1.MachineSet{}).
-		Watches(
-			&clusterv1.MachineSet{},
-			handler.EnqueueRequestsFromMapFunc(r.MachineSetToDeployments),
-		).
```

Comment:
This is only equivalent to Owns if we basically implement the same thing. But MachineSetToDeployments does more than that. I assume because we can't always rely on ownerRefs

Reply:
Makes sense, the labels are there as backup. Going by that logic, now thinking from the opposite perspective wouldn't `Owns()` be redundant?

Reply:
Assuming our implementation is bug free (which should be easy to verify). Yes!